### PR TITLE
AP_Logger: Prepare the maximum number of log files in the config parameter

### PR DESCRIPTION
**Pull request by muramura**
I am using 128 G with micro SD card.
It is inconvenient for me to fix a maximum of 500 log files.
Depending on the micro SD card size, I think that it is better to change it.
I know that SD card is FAT32. For FAT32, 500 is not a limit.

### CODE REVIEW

File: libraries/DataFlash/DataFlash_File.h [hunk 1]
```diff
@@ -103,6 +103,8 @@ class DataFlash_File : public DataFlash_Backend
     ByteBuffer _writebuf;
     const uint16_t _writebuf_chunk;
     uint32_t _last_write_time;
+    //
+    DataFlash_Class *_dataflash;
```
Comment (peterbarker) [inline libraries/DataFlash/DataFlash_File.h]: This is definitely not right.

Check the way we do disarm-rotate on the file backend (`_front`)

**Review by peterbarker — DISMISSED** [review]

**Comment by muramura** [comment]
@peterbarker san... thank you for your comment.
I reflected the comments.

**Comment by rmackay9** [comment]
I think it would be very helpful if we allowed users to limit the number of log files to a small number like 100.  MP seems to struggle when the number is too large and for most users, it's only the last 30 or so that are useful.

**Comment by peterbarker** [comment]
I've rebased this PR on master.

Sadly, it does either introduce or expose problems:
```
ARMED
disarm
STABILIZE> APM: Disarming motors
Got COMMAND_ACK: COMPONENT_ARM_DISARM: ACCEPTED
DISARMED
log list
STABILIZE> Requesting log list
Log 1  numLogs 5 lastLog 5 size 83968 Thu Sep 10 13:49:12 2020
Log 2  numLogs 5 lastLog 5 size 126976 Thu Sep 10 13:49:17 2020
Log 3  numLogs 5 lastLog 5 size 157696 Thu Sep 10 13:49:25 2020
Log 4  numLogs 5 lastLog 5 size 0 
Log 5  numLogs 5 lastLog 5 size 0 

STABILIZE> 
STABILIZE> log list
STABILIZE> Requesting log list
Log 1  numLogs 5 lastLog 5 size 83968 Thu Sep 10 13:49:12 2020
Log 2  numLogs 5 lastLog 5 size 126976 Thu Sep 10 13:49:17 2020
Log 3  numLogs 5 lastLog 5 size 157696 Thu Sep 10 13:49:25 2020
Log 4  numLogs 5 lastLog 5 size 0 
Log 5  numLogs 5 lastLog 5 size 0 
arm throttle
STABILIZE> APM: Arming motors
Got COMMAND_ACK: COMPONENT_ARM_DISARM: ACCEPTED
ARMED
log list
STABILIZE> Requesting log list
APM: Disarming motors
DISARMED
log list
STABILIZE> Requesting log list
Log 1  numLogs 5 lastLog 5 size 126976 Thu Sep 10 13:49:17 2020
Log 2  numLogs 5 lastLog 5 size 157696 Thu Sep 10 13:49:25 2020
Log 3  numLogs 5 lastLog 5 size 0 
Log 4  numLogs 5 lastLog 5 size 0 
Log 5  numLogs 5 lastLog 5 size 0 
```

... but I can't think this one could be too far off.

**Comment by davidbuzz** [comment]
@rmackay9  - last chance to say anything before its closed... do you like the idea of this PR enough to keep it open and encourage @peterbarker  or other/s to rebase and test..?

**Comment by rmackay9** [comment]
@davidbuzz, yes, I think it would be useful to be able to limit the number of log files.  My main sponsor actually asked for this feature as well.

**Pull request closed without merging**